src/main/java/co/elastic/support/rest/RestEntryConfig.java

```diff
@@ -74,24 +74,35 @@ private RestEntry buildRestEntryForVersion(String name, Map<String, Object> entr
             if (semver.satisfies(urlVersion.getKey())) {
                 if (urlVersion.getValue() instanceof String) {
                     return new RestEntry(name, subdir, extension, retry, (String) urlVersion.getValue(), showErrors);
-                    // We allow it to be String,String or String,Map(url,paginate,spaceaware)
+
                 } else if (urlVersion.getValue() instanceof Map) {
                     Map<String, Object> info = (Map<String, Object>) urlVersion.getValue();
-
                     String url = (String) ObjectUtils.defaultIfNull(info.get("url"), null);
-
                     if (url == null) {
                         throw new RuntimeException("Undefined URL for REST entry (route)");
                     }
-
                     String pageableFieldName = (String) ObjectUtils.defaultIfNull(info.get("paginate"), null);
                     boolean spaceAware = (boolean) ObjectUtils.defaultIfNull(info.get("spaceaware"), false);
 
-                    return new RestEntry(name, subdir, extension, retry, url, showErrors, pageableFieldName, spaceAware);
+                    // Construct the RestEntry object first
+                    RestEntry restEntry = new RestEntry(name, subdir, extension, retry, url, showErrors, pageableFieldName, spaceAware);
+
+                    // Logic for handling extra-headers, if they exist, after RestEntry is created
+                    if (info.containsKey("extra-headers")) {
```

Comment:
Is there a downside to just extracting the headers during creation, like we do with `spaceAware` for example?

Reply:
@111andre111 Is there a reason for this or just how you implemented it at the time (either way is fine, I'm just trying to understand)?